lib/index.js

```diff
@@ -112,15 +125,31 @@ function doDownload(stream, url, options, tryCount) {
     if (res.statusCode !== 200 && res.statusCode !== 206) {
       if (redirectCodes.has(res.statusCode)) {
         // Redirection header.
-        doDownload(stream, res.headers.location, options, tryCount - 1);
+        doDownload(stream, res.headers.location, options, tryCount - 1, reconnectInfo);
         return;
       }
       stream.emit('response', res);
       stream.emit('error', new Error('Status code ' + res.statusCode));
       return;
     }
 
-    res.pipe(stream);
+    reconnectInfo.range.end = reconnectInfo.range.end === -1 ? parseInt(res.headers['content-length']) : (parseInt(res.headers['content-length']) < reconnectInfo.range.end ? parseInt(res.headers['content-length']) : reconnectInfo.range.end);
+    res.on('data', function(d) {
+      reconnectInfo.downloaded += d.length;
+      stream.emit('progress', reconnectInfo.downloaded / (reconnectInfo.range.end - reconnectInfo.range.start), reconnectInfo.range.start, reconnectInfo.range.end, reconnectInfo.downloaded, d.length);
```

Comment:
not hurting since u can just ignore them in the listener if u only care about the percentage